weblate/machinery/netease.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright ©2018 Sun Zhigang <hzsunzhigang@corp.netease.com>
+#
+# This file is part of Weblate <https://weblate.org/>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _, ugettext, pgettext
+
+from weblate.machinery.base import (
+    MachineTranslation, MachineTranslationError, MissingConfiguration
+)
+
+import calendar
+import datetime
+import random
+from hashlib import sha1
+
+# Map codes used by Netease Sight to codes used by Weblate
+LANGUAGE_MAP = {
+    'zh_Hans': 'zh',
+}
+
+NETEASE_API_ROOT = 'https://jianwai.netease.com/api/text/trans'
+
+
+class NeteaseSightTranslation(MachineTranslation):
+    """Netease Sight API machine translation support."""
+    name = 'Netease Sight'
+    max_score = 90
+
+    def __init__(self):
+        """Check configuration."""
+        super(NeteaseSightTranslation, self).__init__()
+        if settings.MT_NETEASE_KEY is None:
+            raise MissingConfiguration(
+                'Netease Sight Translate requires app key'
+            )
+        if settings.MT_NETEASE_SECRET is None:
+            raise MissingConfiguration(
+                'Netease Sight Translate requires app secret'
+            )
+
+    def convert_language(self, language):
+        """Convert language to service specific code."""
+        if language in LANGUAGE_MAP:
+            return LANGUAGE_MAP[language]
+
+        return language
```

Comment:
I've factored out this code in 73548f441a21a690cee47da8b64e193528e3da68, now all you need to do is to set language_map object attribute.